src/Mod/Sketcher/App/SketchObject.cpp

```diff
@@ -3320,159 +3388,51 @@ int SketchObject::trim(int GeoId, const Base::Vector3d& point)
-    // (or delete it altogether if it's not on any curve)
-    // Assume that GeoId1 is the pre-existing curve
-    // GeoId2 can be same as GeoId1 (for example for periodic B-spline)
-    auto chooseCurveForPointOnObject = [this](int GeoId1,
-                                              double curve1Start,
-                                              double curve1End,
-                                              int GeoId2,
-                                              double curve2Start,
-                                              double curve2End,
-                                              bool deleteIfOutside = true) {
-        const Part::GeomCurve* curve = static_cast<const Part::GeomCurve*>(this->getGeometry(GeoId1));
-        const std::vector<Constraint*>& constraints = this->Constraints.getValues();
-        std::vector<Constraint*> newConstraints(constraints);
-        int constrId = 0;
-        std::vector<int> delete_list;
-        bool changed = false;
-        for (const auto* constr : constraints) {
-            // There is a preexisting PointOnObject constraint, see where it belongs
-            if (constr->Type == Sketcher::PointOnObject && constr->Second == GeoId1) {
-                Base::Vector3d pp = getPoint(constr->First, constr->FirstPos);
-                double u;
-                curve->closestParameter(pp, u);
-                if ((curve2Start <= u) && (u <= curve2End)) {
-                    std::unique_ptr<Constraint> constrNew(newConstraints[constrId]->clone());
-                    constrNew->Second = GeoId2;
-                    Constraint* constrPtr = constrNew.release();
-                    newConstraints[constrId] = constrPtr;
-                    changed = true;
-                }
-                else if (deleteIfOutside && !((curve1Start <= u) && (u <= curve1End))) {
-                    // TODO: Not on any of the curves. Delete?
-                    delete_list.push_back(constrId);
-                    changed = true;
-                }
-
-            }
-            ++constrId;
-        }
-
-        if (changed) {
-            this->Constraints.setValues(std::move(newConstraints));
-            delConstraints(delete_list, false);
-        }
+        // if (!(constr->Type == Sketcher::Tangent
+        //       || constr->Type == Sketcher::Perpendicular)) {
+        //     return;
+        // }
+
+        // if (constr->First == GeoId1 && constr->Second == GeoId) {
+        //     constr->Type = constr->Type;
+        //     if (secondPos == Sketcher::PointPos::none)
+        //         secondPos = constr->FirstPos;
+        //     delete_list.push_back(constrId);
+        // }
+        // else if (constr->First == GeoId && constr->Second == GeoId1) {
+        //     constr->Type = constr->Type;
+        //     if (secondPos == Sketcher::PointPos::none)
+        //         secondPos = constr->SecondPos;
+        //     delete_list.push_back(constrId);
+        // }

```

Comment:
same comment as other commented out code

Reply:
Removed. I want to keep this method in the future to remove complexity from elsewhere, so just left a `TODO` for now.